doc/source/whatsnew/v0.23.0.txt

```diff
@@ -984,6 +984,8 @@ I/O
 - Bug in :meth:`pandas.io.stata.StataReader.value_labels` raising an ``AttributeError`` when called on very old files. Now returns an empty dict (:issue:`19417`)
 - Bug in :func:`read_pickle` when unpickling objects with :class:`TimedeltaIndex` or :class:`Float64Index` created with pandas prior to version 0.20 (:issue:`19939`)
 - Bug in :meth:`pandas.io.json.json_normalize` where subrecords are not properly normalized if any subrecords values are NoneType (:issue:`20030`)
+- Bug in :func:`read_excel` and :class:`TextReader` now turn np.nan to empty string when dtype=str. They used to turn np.nan to 'nan' (:issue `20377`)
```

Comment:
Perhaps phrase this as "Bug in read_excel, old behavior, new behavior". Right now it reads a bit like turning NaN to the empty string was the bug.

TextReader isn't part of the public API so this link won't work (and it'll cause a warning in the doc build.)

Double backticks around `np.nan`.

Double backticsk around `dtype=str`.

Double backticks around `'nan'` (keep the quotes though).

Reply:
Should also mention that this affected parsing CSVs with `dtype=str`, right?:

Reply:
> Should also mention that this affected parsing CSVs with dtype=str, right?

@TomAugspurger : yes